src/target/riscv/riscv.c

```diff
@@ -626,6 +625,58 @@ static int maybe_add_trigger_t2(struct target *target,
 	return ERROR_OK;
 }
 
+static int maybe_add_trigger_t6(struct target *target,
+		struct trigger *trigger, uint64_t tdata1)
+{
+	RISCV_INFO(r);
+
+	/* tselect is already set */
+	if (tdata1 & (CSR_MCONTROL6_EXECUTE | CSR_MCONTROL6_STORE | CSR_MCONTROL6_LOAD)) {
+		/* Trigger is already in use, presumably by user code. */
+		return ERROR_TARGET_RESOURCE_NOT_AVAILABLE;
+	}
+
+	/* address/data match trigger */
+	tdata1 |= MCONTROL_DMODE(riscv_xlen(target));
+	tdata1 = set_field(tdata1, CSR_MCONTROL6_ACTION,
+			MCONTROL_ACTION_DEBUG_MODE);
+	tdata1 = set_field(tdata1, CSR_MCONTROL6_MATCH, MCONTROL_MATCH_EQUAL);
+	tdata1 |= CSR_MCONTROL6_M;
+	if (r->misa & (1 << ('H' - 'A')))
```

Comment:
and trigger type 6 is new in spec too, and includes new VS and VU fields (along with other small bitwise adjustments); got it.  Looks good to me.